### PR TITLE
feat: add configuration entry for TLS termination at HA Proxy

### PR DESCRIPTION
**Pull request by wyattrees**
This PR adds a `tls_mode` configuration option, which determines how HA Proxy's `services` yaml configuration should be updated. TLS Passthrough will come in a second PR.

This version is released to channel `latest/edge/wyatt-test`

### CODE REVIEW

File: maas-region/src/charm.py [hunk 1]
```diff
@@ -264,8 +274,25 @@ def _update_ha_proxy(self) -> None:
...
+                            (
+                                f"{app_name}-{self.unit.name.replace('/', '-')}",
+                                self.bind_address,
+                                MAAS_HTTP_PORT,
+                                [],
+                            )
+                        ],
+                    }
+                )
+            # TODO: Implement passthrough configuration
```
Comment (alexsander-souza) [inline maas-region/src/charm.py:295]: raise an exception if this mode can be selected

File: maas-region/src/charm.py [hunk 2]
```diff
@@ -55,6 +56,12 @@
 class MaasRegionCharm(ops.CharmBase):
     """Charm the application."""
 
+    _TLS_MODES = [
+        "",
+        "termination",
+        "passthrough",
```
Comment (alexsander-souza) [inline maas-region/src/charm.py:62]: either remove this missing mode or raise an exception when the user selects it. we should not allow the user to set an option that does nothing and fails silently.

File: maas-region/tests/unit/test_charm.py [hunk 3]
```diff
@@ -118,6 +119,22 @@ def test_ha_proxy_data(self, mock_helper):
         self.assertEqual(len(ha_data[0]["servers"]), 1)
         self.assertEqual(ha_data[0]["servers"][0][1], "10.0.0.10")
 
+    @patch("charm.MaasHelper", autospec=True)
+    def test_ha_proxy_data_tls(self, mock_helper):
```
Comment (alexsander-souza) [inline maas-region/tests/unit/test_charm.py:123]: add tests for unsupported/invalid modes

**Review by alexsander-souza — CHANGES_REQUESTED** [review]

File: maas-region/src/charm.py (hunk 1 above)
Comment (wyattrees) [reply maas-region/src/charm.py:295]: opting with removing "passthrough" as a valid option

**Review by alexsander-souza — APPROVED** [review]
+1

**Pull request merged**